### ISSUE TITLE
Confusion about the initial evaluation results of the GPT2-LARGE model

### ISSUE THREAD

**Issue by Lisennlp** [facebookresearch/MetaICL#19]
Hello, I have a puzzle, I use gpt2-large pre-training language model to evaluate unseen_domain_test in-context, config file is configs/class_to_class.json, method is direct, k=16, found in-context f1 result is 38.63( the paper‘result is 30.6) 。in the paper or config file, we can know the unseen_domain_test dataset include "poem_sentiment", "climate_fever", "medical_questions_pairs", "financial_phrasebank" , all is  2 categories or 3 categories, even if a random pick should be at least 33.3(1/3), why is the paper 30.6? A little confused.

**Comment by shmsw25** [comment]
Hi, thanks for your interest to our work!

First, I just double-checked reproducing the ICL results using GPT-2 Large, and still gets the number reported in the paper (30.6). Could you verify the command line you ran? The correct command line is as follows:

```bash
python test.py --task class_to_class --unseen_domain_only --k 16 --seed 100,13,21,42,87 --split test --method direct --use_demonstrations --out_dir out/gpt2 --do_zeroshot
```

RE random baseline: Yes, we double-checked and found ICL results are worse than the random guessing baselines, likely because these four datasets are in a specific domain that GPT-2 Large does not perform well, e.g., climate-fever, and medical_questions_pairs. This is possible, especially given that GPT-2 Large is a fairly small model.

**Comment by Lisennlp** [comment]
Thank you for your reply! 
I got a 38.6 rating based on the script I wrote for your paper, which is a bit different. I'm going to try nothing at all.
I noticed that you added the --do_zeroshot parameter to your command. So I re-evaluated, following your command:
`CUDA_VISIBLE_DEVICES=1  python test.py --task class_to_class --unseen_domain_only --k 16 --seed 100,13,21,42,87 --split test --method direct --use_demonstrations --out_dir out/gpt2 --do_zeroshot --test_batch_size 1`
The result is 28.5 which closely to your result.
<img width="1434" alt="image" src="https://user-images.githubusercontent.com/43019168/208572265-859245e4-f4a9-42d5-bb02-e4e19ed8c15d.png">

Then, I also evaluated it following the command:
`CUDA_VISIBLE_DEVICES=1 python test.py --task class_to_class --k 16 --split test --seed 100,13,21,42,87 --test_batch_size 1 --method direct --use_demonstrations --out_dir  ./checkpoints/class_to_class/ --global_step 0 --unseen_domain_only`
and found that the results were really bad(22.0), but much worse than 30.6. 
'./checkpoints/class_to_class/ '  dir have the Model I pre-download, and I changed the name to model-0.pt
<img width="1284" alt="image" src="https://user-images.githubusercontent.com/43019168/208568984-a17afe13-9c1a-4a3e-8806-c53f91dd7176.png">

The code is cloned from your github library，Am I missing something？

I used the following environment:
- pytorch1.11
- python3.9
- cuda11.4

**Comment by shmsw25** [comment]
The version information is in [README](https://github.com/facebookresearch/MetaICL#installation). 

Sorry, I'm not exactly sure why you weren't able to get the exact numbers in the paper, especially given that I and a few others (like a user in [this issue](https://github.com/facebookresearch/MetaICL/issues/5)) have successfully reproduced the results in the paper. Maybe try using the exactly same version, and if the results are still slightly off, share the log file with me so that we can see which datasets are causing the gap.

From two datasets shown in your image, it looks like `medical_questions_pairs` (s=87) result is the same as mine, but `financial_phrasebank` result is a bit different (mine if 0.4518385).

(It is possible that the version of the datasets is different. Please double-check if you are using the same version as specified [here](https://github.com/facebookresearch/MetaICL#how-to-download-and-preprocess) in README.)

**Comment by Lisennlp** [comment]
ok, thanks for your patient reply, I will check code and datasets and try it again, thanks!
